src/prompt/variable.rs

```diff
@@ -16,31 +35,200 @@ use rustyline::hint::Hinter;
 use rustyline::validate::{Validator, ValidationResult, ValidationContext};
 use rustyline::{Helper, Context};
 
+type ParseResult<'a, I = &'a str, R = Value> = IResult<I, R, ParsingError>;
 
 #[derive(Debug, thiserror::Error)]
-pub enum Error {
+pub enum ParsingError {
     #[error("{}", description)]
-    Mistake { offset: Option<usize>, description: String },
+    Mistake { kind: Option<ErrorKind>, description: String },
```

Comment:
No, `ParsingError` cant directly calculate the offset since its constructed with only the context of the current slice. An offset can be calculated later when we format `ParsingError`, which sounds like a good idea.